azext_edge/edge/util/x509.py

```diff
@@ -73,3 +73,13 @@ def generate_self_signed_cert(valid_days: int = DEFAULT_VALID_DAYS) -> Tuple[byt
     )
 
     return (cert.public_bytes(serialization.Encoding.PEM), key_bytes)
+
+
+def decode_der_certificate(der_data) -> x509.Certificate:
```

Comment:
This should be `Optional[x509.Certificate]`

Reply:
Re-opened this because I'm wondering about `der_data: bytes`, is adding bytes type hint causing an issue?

Reply:
no, my eyes missed something. will update!😋